pkg/secrethub/secret.go

```diff
@@ -13,7 +13,28 @@ type SecretService interface {
 	Exists(path string) (bool, error)
 	// Get retrieves a Secret.
 	Get(path string) (*api.Secret, error)
+	// EventIterator returns an iterator that retrieves all audit events for a given secret.
+	//
+	// Usage:
+	//  iter, err := client.Secrets().EventIterator(path)
+	//  if err != nil {
+	// 	    // Handle error
+	//  }
+	//  for {
+	// 	    event, err := iter.Next()
+	//	    if err == secrethub.IteratorDone {
+	//	        break
+	//	    }
+	//	    if err != nil {
+	//	        // Handle error
+	//	    }
+	//
+	//      // Use event
+	//  }
```

Comment:
Personally I don't feel the usage has become any clearer by our abstraction. We now have three error checks that don't all make super much sense intuitively:

- On `.EventIterator`
- At `if err == secrethub.IteratorDone` (and why is that in the `secrethub` package?)
- At `if err != nil`

Not entirely sure what is causing the awkwardness, but it feels to me like one of these cases would be simpler:

```go
iter, err := client.Secrets().EventIterator(path)
for {
   event, err := iter.Next()
   if err == iterator.Done {
      break
   } else if err != nil {
      // Handle error
   }
   // Use event
}
```
Or Stripe flavored:
```go
iter, err := client.Secrets().EventIterator(path)
for iter.HasNext() {
   event := iter.Next()
   // Use event
}
err := iter.Err()
if err != nil {
   // handle error
}
```

Reply:
https://github.com/secrethub/secrethub-go/pull/144#discussion_r340206520